gate_controller/reolink_snapshots.py

```
@@ -0,0 +1,513 @@
+            for sequence in range(1, self._config.candidate_count + 1):
+                response = client.snapshot(
+                    token, sequence, self._remaining(deadline, stop_event)
+                )
+                self._remaining(deadline, stop_event)
+                path = self._store_candidate(prefix, sequence, response)
+                accepted.append(path)
+        except SnapshotFailure as error:
+            reason = error.reason
+        except (OSError, ValueError):
+            reason = "io_error"
+        except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)